src/linked_list.py

```diff
@@ -0,0 +1,69 @@
+"""Implementation of Linked_List data type."""
+
+
+class LinkedList(object):
+    """Class representation of linked list."""
+
+    def __init__(self, head_node):
```

Comment:
Check the assignment description again; the constructor function should optionally take an iterable (e.g. a list).

Your implementation requires the user to pass in a Node instance. This is a very, very, very bad idea.

Consider that you are making this data structure to actually be used. It makes no sense to force the user to import *both* the `LinkedList` and `Node` class, then construct an instance of `Node` in order to even initialize a `LinkedList`. This is very user-unfriendly; no data type in Python works like this. You've done the right thing in your `push` method, which is to construct the `Node` instance within the method.